int16 now_step
int16 x
int16 y
int16 Width
int16 Height
int16 c_x
int16 c_y